L1C_Aux/collocate_hirs_avhrr.py

```diff
@@ -58,143 +65,171 @@ def create_filepaths(args):
+
+
+
+                for g in os.listdir(gac_dir+string_date):
+                    time_gac = rf.read_time(gac_dir+string_date+g)               
+                    #Identify files where the HIRS/AVHRR times overlap
+                    if np.min(time_gac) > np.max(time_hirs):
+                        continue
+                    elif np.max(time_gac) < np.min(time_hirs):
+                        continue
+                    else:
+                        print 'overlap'
+                        #Identify overlap times for GAC and HIRS
+                        gac_min,gac_max,hirs_min,hirs_max,data_check \
+                            = td.time_mask(time_hirs,time_gac)
+                        if data_check == True:
+                        #Read in for GAC and HIRS: lat,lon and l2 flags
+
+                            gac_lat,gac_lon,gac_flags \
+                            = rf.refine_geo(gac_dir+string_date+g,gac_min,gac_max)
+                            hirs_lat,hirs_lon,hirs_flags \
+                            = rf.refine_geo(hirs_dir+string_date_hirs+f,\
+                                             hirs_min,hirs_max)
+
+                        #Read in GAC and HIRS BT
+                            hirs_bt,hirs_noise,hirs_obs,hirs_ind,hirs_str,hirs_com\
+                            = rf.read_hirs_obs(fiduceo_dir+string_date_hirs \
+                                                   +fiduceo_filename,hirs_min,hirs_max)
+
+                            l1c_filename = rf.construct_l1c(l1c_dir,string_date,g,gac_dir)
+                            
+                            # Code now attempts to find file with matching hour start time
+                            try:
+                                os.path.isfile(l1c_dir+string_date+l1c_filename)
+                                #l1c_filename = rf.construct_l1c(l1c_dir,string_date,g,gac_dir)
+                                gac_bt,avhrr_obs,avhrr_noise \
+                                = rf.read_avhrr_obs(l1c_dir+string_date+l1c_filename,\
+                                                    gac_min,gac_max,args.hirs_sensor)
+                                if os.path.isfile(avhrr_sim_dir+string_date+g):
+                                    gac_dy = rf.read_dy(avhrr_sim_dir+string_date+g,gac_min,\
+                                                    gac_max)
+                                else:
+                                    gac_dy=np.zeros([1])
+                                gac_prob = rf.read_pclear(gac_dir+string_date+g,gac_min,gac_max)
+                                
+                            # If hour-matched start time is missed, it now finds the closest in time and replaces 
+                            # l1c_filename with the a new version which contains the time of the nearest file
+                            except:
+                                print "Mismatched file found in l2p and l1c for: ",g
+                                old_g_time = int(g[8:14])
```

Comment:
I would be inclined to move these ~five lines of code to a separate function for clarity.